### PR TITLE
Fix threading bug, lazily create submission service

### PR DESCRIPTION
**Pull request by adam-baserun**
FYI @erik-megarad

### CODE REVIEW

File: baserun/thread_wrapper.py [hunk 1]
```diff
@@ -13,20 +13,16 @@ def baserun_thread_wrapper(target: Callable):
     if not Baserun._initialized:
         return target
 
-    parent_span: _Span = get_current_span()
+    parent_span: Span = get_current_span()
 
     def wrapper(*args, **kwargs):
         tracer_provider = trace.get_tracer_provider()
         tracer = tracer_provider.get_tracer("baserun")
-        span = tracer.start_span(
```
Comment (erik-megarad) [inline baserun/thread_wrapper.py:21]: What's the reason behind this change?

Without the `span.is_recording()` check, when the context manager exits it will `end()` it without checking to see if it's already ended. https://github.com/open-telemetry/opentelemetry-python/blob/34d11d59e9b37052eb3aa6706288641ecb3056e7/opentelemetry-api/src/opentelemetry/trace/__init__.py#L595

Comment (adam-baserun) [reply baserun/thread_wrapper.py:21]: Gotcha I think the bug is that we're not "using" the span in this context so I was seeing issues the downstream span was a NonRecordingSpan and so we couldn't read / write its attributes. 

```
  This is a convenience method for creating spans attached to the
        tracer's context. Applications that need more control over the span
        lifetime should use :meth:`start_span` instead. For example::

            with tracer.start_as_current_span(name) as span:
                do_work()

        is equivalent to::

            span = tracer.start_span(name)
            with opentelemetry.trace.use_span(span, end_on_exit=True):
                do_work()
```

To your point there is probably a version of this where we manually manage the span basically doing 

use_span(span, end_on_exit=False) and then checking if span.is_recording in the finally block as before. Does that sound better?

Comment (adam-baserun) [reply baserun/thread_wrapper.py:21]: Pushed another commit, thoughts?

**Review by erik-megarad — APPROVED** [review]
